### ISSUE TITLE
CPU paged attention reads bf16 KV pool as f32, producing wrong output (GPU path is correct)

### ISSUE THREAD

**Issue by aien-dev** [aien-dev/aien-sovereign-core#97]
## Summary

The default CPU `TensorBackend::paged_attention` reads the KV pool as f32 no matter what dtype the pool uses. When the pool is `KvDType::Bf16`, which is what the runtime end-to-end tests use, it reads the wrong bytes and produces wrong attention output. The GPU kernel (`paged_attention_bf16.cu`) is correct.

Verified at `36280cb` (PR #68 merge) on `spark-b87b` (NVIDIA GB10, driver 580.173.02).

## Location

`crates/aien-inference-abi/src/backend.rs:111-127`

```rust
let head_offset_bytes = kv_head * head_dim * std::mem::size_of::<f32>();
...
std::slice::from_raw_parts(pool.base_ptr().add(k_offset) as *const f32, head_dim)
```

`UnifiedKvTensorPool::write_token_kv` stores bf16 bits as `u16` for `KvDType::Bf16` (`crates/aien-kv-cache/src/lib.rs:487`). `ReferenceCpuBackend` doesn't override this default, and `BlackwellGb10Backend` falls back to it when a GPU call fails.

## Evidence

The same prompt `[5, 12, 33, 77]` was run through `AienRuntimeSpine` and `NativeTransformerBackend` using the `aien-micro-v1` config, `reference_test_weights`, a bf16 pool and greedy sampling:

| Backend | Tokens |
|---|---|
| `BlackwellGb10Backend` | `[246, 8, 78, 21, 110, 243, 179, 89]` |
| `ReferenceCpuBackend` | `[246, 65, 142, 134, 70, 252, 12, 238]` |

The two agree on the prefill token and diverge from the first decode step.

**Per-op isolation.** Each op was switched from CPU to GPU one at a time. `matmul_vec`, `matmul_batch` and `compute_logits` do not change the tokens. `paged_attention` alone reproduces the whole divergence.

**Comparison against an independent bf16-aware reference** (online softmax that decodes the stored bf16 bits correctly), on the live pipeline inputs:

| ctx | layer | \|GPU − bf16 ref\| | \|CPU default − bf16 ref\| |
|---|---|---|---|
| 5 | 0 | 0.00022 | 0.12745 |
| 5 | 1 | 0.00048 | 0.20286 |
| 8 | 1 | 0.00046 | 0.19134 |
| 11 | 1 | 0.00024 | 0.15901 |

With the bf16 reference swapped in, the tokens match the GPU exactly.

## Impact

- `test_end_to_end_ticket_submission_and_channel_streaming_cpu` and `test_end_to_end_subagent_branching_with_independent_sinks` use a bf16 pool with `ReferenceCpuBackend`. They pass while computing incorrect attention, because they check token counts, routing, streaming and block reclamation, not token values.
- If a Blackwell GPU call fails on a bf16 pool, the fallback silently produces wrong numbers rather than just running slower.
- **Possible out-of-bounds read.** For `kv_head >= 1` the byte offset uses 4-byte elements, and the function then reads `head_dim * 4` bytes. That lands in the next head or token and, at the last slot of the last block, likely past the end of the pool allocation. This has not been confirmed with a sanitizer.
- The existing parity tests don't cover this case. `paged_transformer_parity` does not use a bf16 pool, and `paged_attention_parity` compares the kernel against its own separate oracle rather than the trait default.

## Suggested fix

1. Make the default `paged_attention` dtype-aware. Match on `pool.config().dtype`, use the element size for the head offset, and decode bf16 via `bf16_bits_to_f32`.
2. Add a full-pipeline parity test that runs the same prompt through the GPU and CPU backends with a bf16 pool and asserts identical greedy tokens, skipping when no GPU is present. Also add a CPU-only test comparing a bf16 pool against an f32 pool (tokens equal, or logits within tolerance).
3. Optional: have the end-to-end tests assert token values, not just counts.

## Related observations (separate from this bug)

- `fallback_count()` only counts GPU calls that failed. RMSNorm, RoPE, SwiGLU and prefill attention run on CPU by design and are not counted. In the 4-token GB10 run the GPU executed 60 GEMM/GEMV calls and 6 paged-attention calls (2 layers × 3 decode steps).
- `kernel_exec_count()` reads process-global counters in the CUDA code, not per-instance counters.
- `paged_attention_bf16_forward` copies the entire KV pool to the device on every call when the pool is host memory (`paged_attention_bf16.cu`, around line 337). The cost grows with pool size × layers × steps.

## Reproduction

The probe test file (`crates/aien-runtime/tests/zz_sentinel_probe.rs`) runs with `cargo test -p aien-runtime --test zz_sentinel_probe -- --nocapture` on GB10. It is not committed; it is available on `spark-b87b` at `/tmp/opencode/pr68-artifacts/`.

**Comment by aien-dev** [comment]
The CPU paged-attention path now decodes KvDType::Bf16 through bf16_bits_to_f32 in crates/aien-inference-abi/src/backend.rs. That landed on main with #99, merge 9cec202.
